insights/parsers/transparent_hugepage.py

```diff
@@ -0,0 +1,85 @@
+"""
+transparent_hugepage sysfs settings
+===================================
+
+Module for parsing the sysfs settings for transparent_hugepage:
+
+ThpUseZeroPage - file ``/sys/kernel/mm/transparent_hugepage/use_zero_page``
+---------------------------------------------------------------------------
+
+Gets the contents of /sys/kernel/mm/transparent_hugepage/use_zero_page, which is either 0 or 1.
+
+Sample input::
+
+    0
+
+Examples:
+
+    >>> shared[ThpUseZeroPage].get()
+    0
+
+
+ThpEnabled - file ``/sys/kernel/mm/transparent_hugepage/enabled``
+-----------------------------------------------------------------
+
+Gets the contents of  /sys/kernel/mm/transparent_hugepage/enabled, which is something like
+`always [madvise] never` where the active value is in brackets.
+
+If no option is active (that should never happen), `get_opt()` will return None.
+
+Sample input::
+
+    always [madvise] never
+
+Examples:
+
+    >>> shared[ThpEnabled].get_line()
+    always [madvise] never
+    >>> shared[ThpEnabled].get_opt()
+    madvise
+
+"""
+
+from .. import Parser, parser
+
+
+@parser("thp_use_zero_page")
+class ThpUseZeroPage(Parser):
+    """
+    Gets the contents of /sys/kernel/mm/transparent_hugepage/use_zero_page, which is either 0 or 1.
+    """
+    def parse_content(self, content):
+        self.data = " ".join(content).strip()
+
+    def get(self):
+        """Returns the string value."""
+        return self.data
+
+
+@parser("thp_enabled")
+class ThpEnabled(Parser):
+    """
+    Gets the contents of  /sys/kernel/mm/transparent_hugepage/enabled, which is something like
+    `always [madvise] never` where the active value is in brackets.
+    If no option is active (that should never happen), get_opt() will return None.
+    """
+    def parse_content(self, content):
+        self.data = " ".join(content).strip()
+        self.active_option = None
+        for w in self.data.split():
+            if w.startswith("[") and w.endswith("]"):
+                self.active_option = w[1:-1]
+
+    def get_line(self):
+        """
+        Returns the string value of the whole output.
+        E.g. `always [madvise] never`
+        """
+
+        return self.data
+
+    def get_opt(self):
```

Comment:
@jsvob have you considered using attributes or properties for all of these methods in both classes?

Reply:
No. Can you please point me to an existing parser that uses it in the way you'd like me to implement it?

Reply:
See my examples on above class.

Reply:
I ended up using attributes because when I tried properties, the names of attributes and properties collided anyway.